Add helping section

itscynxx opened the pull request:
Pretty sure this is my biggest PR yet, the time it takes to get fully merged will probably reflect this 

There are a few parts I'm iffy on (notably the hyperlink on line 100. Though for this one, hovering over the first [ shows that it's "linked" to the proper ] for the image so it might work)

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,142 @@
+# Helping
+
+This section is meant mostly for those that will be actively helping others in the Northstar Discord server or otherwise. If you've somehow stumbled upon this page trying to fix an issue for yourself, you should try checking the [troubleshooting page](installing-northstar/troubleshooting.md). If you're a helper, it also isn't a bad idea to look at this page a few times and try to have a rough ide ...
```
I'd say have line breaks between sentences in the markdown source. Those don't get rendered as line breaks on GitBook (cause that's markdown convention) but make it easier to suggest changes to a single sentence cause it results in a smaller diff (sentence vs paragraph).

In other words

```suggestion
This section is meant mostly for those that will be actively helping others in the Northstar Discord server or otherwise.
If you've somehow stumbled upon this page trying to fix an issue for yourself, you should try checking the [troubleshooting page](installing-northstar/troubleshooting.md).
If you're a helper, it also isn't a bad idea to look at this page a few times and try to have a rough idea of the things that are on it.
```

This isn't the only place, just the first I found ^^

GeckoEidechse commented on docs/SUMMARY.md:
```diff
@@ -55,5 +55,5 @@
 
 ## Other
-
+* [Helping](helping.md)
```
We probably wanna keep the spacer line to be consistent with the rest of the file.

```suggestion

* [Helping](helping.md)
```

GeckoEidechse reviewed (changes requested):
Gonna leave it at the line break stuff for now cause if suggest any changes and you push line breaks first, all my suggestions will merge conflict and cannot be auto-applied anymore ^^

GeckoEidechse commented on docs/helping.md:
```diff
@@ -1,38 +1,55 @@
 
-"How do I (manually) install Northstar?": `/tag id:manual-install` (This goes over installing, extracting, and replacing the files as well as giving a video guide. Spectre should catch _most_ of these questions)
+"How do I (manually) install Northstar?": `/tag id:manual-install` (This goes over installing, extracting, and replacing the files as well as giving a video guide. 
+Spectre should catch _most_ of these questions)
```
Eh maybe avoid Spectre mention until it's added. Also when adding, might make sense to mention somewhere what _Spectre_ is ^^

itscynxx replied:
this may or may not have been left over from an original commit where i was assuming id be working on this for weeks and Spectre would've been added already, there was a section explaining Spectre but it's since removed

itscynxx replied:
The original explanation:
> As of {ENTER DATE HERE WHEN SPECTRE GETS ADDED}, Cyn(cooldudepugs)'s Discord bot [Spectre](https://github.com/CooldudePUGS/Spectre) was added to Northstar's Discord. This bot tries to automatically redirect users to opening a ticket in the `#help` channel, and automatically replies to some messages regarding issues with the game, however it cannot and will not catch all cases, even of basic help.

Would probably edit this to say 
> As of {ENTER DATE HERE WHEN SPECTRE GETS ADDED}, Cyn(cooldudepugs)'s Discord bot [Spectre](https://github.com/CooldudePUGS/Spectre) was added to the Northstar Discord. This bot tries to automatically redirect users to opening a ticket in the `#help` channel, and automatically replies to some messages regarding issues or questions with Northstar. It, however, cannot and will not catch all cases, even of basic help.

GeckoEidechse commented on docs/helping.md:
```diff
@@ -74,8 +94,10 @@ Note that `close_delay` is a number, meaning the time in hours before the ticket
 If the ticket very highly seems resolved/the user has said they have no issues, use `/closerequest` with `close_delay` set to `1` and the `reason` set to "resolved, ac 1hr"\
-If the ticket seems like it _might_ be resolved, try a `close_delay` of `3-8` (use your judgement) with the `reason` "seemingly resolved, ac {HOUR-COUNT}hr". This gives the user more time to deny the request if the issue isn't actually resolved\
-For inactive tickets (when it's the _user_ not responding) we usually give them a few days (3-5, sometimes longer if we forget) then start pinging them once a day. If they continue to not respond for 2-3 days, we tend to set a close request with `close_delay` set to `24` or `48` (`24` is especially nicer for users who _never_ responded to their ticket), and the `reason` set to "no response, ac {H ...
+If the ticket seems like it _might_ be resolved, try a `close_delay` of `3-8` (use your judgement) with the `reason` "seemingly resolved, ac {HOUR-COUNT}hr". 
```
I'd keep 8 hour auto close the minimum for resolved ^^

itscynxx replied:
do you mean _all_ resolved or seemingly resolved?

GeckoEidechse replied:
I'd say all resolved. The ticket author can still manually resolve earlier if they wanna ^^

GeckoEidechse commented on docs/helping.md:
```diff
@@ -74,8 +94,10 @@ Note that `close_delay` is a number, meaning the time in hours before the ticket
 Using "ac" as described later lets other people know how long until the ticket will close. "ac" meaning **A**uto **C**lose.
```
Maybe mention that it's cause ticket bot doesn't mention the time a ticket was set to autoclose at. ^^

GeckoEidechse reviewed (commented):
So much to read so I just did part of it for now ^^"

itscynxx commented:
me when i forget to check github for 3 days

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,188 @@
+# Helping
+
+This section is meant mostly for those that will be actively helping others in the Northstar Discord server or otherwise.
+If you've somehow stumbled upon this page trying to fix an issue for yourself, you should try checking the [troubleshooting page](installing-northstar/troubleshooting.md). 
```
```suggestion
If you've somehow stumbled upon this page trying to fix an issue for yourself, you should try checking the [troubleshooting page](installing-northstar/troubleshooting.md) instead.
```

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,188 @@
+When helping a user, please try to move them to the `#help` channel to create a ticket as well as to keep public channels less cluttered. 
+Cyn (cooldudepugs)'s Discord bot [Spectre](https://github.com/CooldudePUGS/Spectre) is meant to additionally try to direct users to tickets as well when it detects a user might be asking for help in a public channel. It also automatically replies to specific error messages, which you can see a list of [here](https://github.com/CooldudePUGS/Spectre/wiki/Replies).
+If they really don't want to open a ticket or you don't want to deal with a ticket, ask them to move to `#northstar-chat`. 
+There, not only can they embed things, but it'll be slightly more focused on helping than general would be.
```
```suggestion
There, not only can they embed things, but it'll be slightly more focused on helping than other channels would be.
```

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,188 @@
+There, not only can they embed things, but it'll be slightly more focused on helping than general would be.
+
+Arguably the most important things to know as a helper are the ability to read logs and deal with users that don't read the wiki (90% of tickets are solutions solvable by checking the wiki.
+There is a pretty high chance that even if the user says yes, they read the wiki, that they didn't or didn't find their issue on there)
```
```suggestion
There is a pretty high chance that even if the user claims they read the wiki they didn't actually do so or didn't find their issue on there)
```

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,188 @@
+There is a pretty high chance that even if the user says yes, they read the wiki, that they didn't or didn't find their issue on there)
+
+Reading logs is a little tougher however, but once you get the basics down you can easily diagnose issues in a few seconds after opening a log. 
+You can use `/tag id:logs` to inform users how to properly send a log in a Discord channel. Generally, when reading a log there are a list of things you want to look for, which will be covered later.
```
```suggestion
You can use `/tag id:logs` to inform users how to properly send a log in a Discord channel.
Generally, when reading a log there are a list of things you want to look for, which will be covered later.
```

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,188 @@
+
+The first section will first go over extremely common issues and their tags, then the basics of using the ticket bot for commands inside tickets, and then some overview and specific examples for reading logs.
+
+Additionally, please don't be afraid to ask for help. The entirety of `#volunteers-chat` in the Discord server is specifically for those that help/sometimns help users and there's no shame it not knowing.
```
```suggestion
Additionally, please don't be afraid to ask for help. The entirety of `#volunteers-chat` in the Discord server is specifically for those that help/sometimes help users and there's no shame it not knowing.
```

GeckoEidechse reviewed (commented):
Partial review again. This PR is so much text that I tend to just drift off halfway through...

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,188 @@
+The tag command for the tickets bot is `/tag`, and you can either type `/tag id:{TAG ID HERE}` or let the bot autofill the tag for you after typing `/tag` and selecting the tag id from the list (make sure you select the ticket bot, and not Dyno!). 
+Note, you can't reply to a user while using a slash command, so try to reply to the bot's message after sending the tag command while pinging the person it's directed at and informing them that they should follow the provided solution.
+
+Here, this guide will go over common issues and their tag id's, trying to have the more commonly used tags near the top.
```
This list will get outdated super fast when we add/remove tags. Is there no way to see list of tags and their function in ticket bot (dashboard). Cause then we should just link to that IMO.

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,188 @@
+   When reading a log, the errors are generally easy to find. You can usually scroll to the bottom of the log, then look for an error message for script compilation errors, or a crash dump.
+   Note that the crash dump isn't usually all to helpful in itself, however for both crash dumps and script compilations, generally you can look at the areas surrounding it to see what mod is loaded before the error and start from there. Read the specifics for more in depth explanations of each.
+   
+3. Time and date
```
Where's `2.`? ^^"

GeckoEidechse commented:
I'm starting to think we might wanna move the log reading part into a separate PR ^^"

itscynxx commented:
Updated, removed the log reading section, fixed some typos, added mentions of both Discord bots to a separate bit of the guide (because I don't really want to shove it in people's faces, but it is nice to know), as well as moving the tags mention to a link to the help bot's dashboard. It just links to `dashboard/<NORTHSTAR_SERVER_ID>/tags`, meaning users still need to log in with permissions to view them.

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,85 @@
+If you've somehow stumbled upon this page trying to fix an issue for yourself, you should try checking the [troubleshooting page](installing-northstar/troubleshooting.md) instead. 
+If you're a helper, it also isn't a bad idea to look at this page a few times and try to have a rough idea of the things that are on it.
+
+When helping a user, please try to move them to the `#help` channel to create a ticket as well as to keep public channels less cluttered. 
```
```suggestion
When helping a user, please try to move them to the `#help` channel to create a ticket in order to keep public channels less cluttered. 
```

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,85 @@
+There, not only can they embed things, but it'll be slightly more focused on helping than other channels would be.
+
+Arguably the most important things to know as a helper are the ability to read logs and deal with users that don't read the wiki (90% of tickets are solutions solvable by checking the wiki.
+There is a pretty high chance that even if the user claims they read the wiki they didn't actually do so or didn't find their issue on there)
```
```suggestion
There is a pretty high chance that even if the user claims they read the wiki they didn't actually do so or didn't manage to find their issue on there)
```

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,85 @@
+
+## Common issues and their tags
+
+In the Northstar Discord server, a massively used device to help end users is the tickets bot. A lot of the common issues and questions that come up have tags for them on the Discord server. 
```
Might wanna take a second to explain what a `tag` is in the context of ticket bot.

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,85 @@
+You can use `/tag id:logs` to inform users how to properly send a log in a Discord channel. 
+Generally, when reading a log there are a list of things you want to look for, which will be covered later.
+
+The first section will first go over extremely common issues and their tags, then the basics of using the ticket bot for commands inside tickets, and then some overview and specific examples for reading logs.
```
The example for reading logs is gone, no?

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,85 @@
+This entire section only applies to users with `ticket-staff`+ roles who can see the ticket channels in the Discord server
+
+The tickets have a fairly straightforward way of operating. When a user clicks on a button to open a ticket 
+(a staff ticket is only visible by helper+ roles, while a public ticket is visible by ticket-staff+ roles) they're greeted with the following screen:\
```
Uh the stuff in parenthesis seems like it's kinda coming out of nowhere ^^"

GeckoEidechse reviewed (commented):
Another unfinished review.

Honestly there's so much text in this that it will be an endless back and forth. It might make more sense to first do a PR that adds the structure and then PR the sections one by one slim things down.

(If we do that, please keep this PR here open though so that we can see the progress in the diff etc)

GeckoEidechse commented on docs/helping.md:
```diff
@@ -0,0 +1,85 @@
+The tag command for the tickets bot is `/tag`, and you can either type `/tag id:{TAG ID HERE}` or let the bot autofill the tag for you after typing `/tag` and selecting the tag id from the list (make sure you select the ticket bot, and not Dyno!). 
+Note, you can't reply to a user while using a slash command, so try to reply to the bot's message after sending the tag command while pinging the person it's directed at and informing them that they should follow the provided solution.
+
+You can find a list of these tags on the [tag dashboard for the help bot](https://panel.ticketsbot.net/manage/920776187884732556/tags), so long as you sign in using your Discord account and have access to it with the roles that you have.
```
This link causes linkcheck to fail due to missing access. The easiest solution is to add it to its ignore list under `.github/.markdownlinkcheck.json`

GeckoEidechse commented on docs/helping.md:
```diff
@@ -1 +1,85 @@
+
+The Northstar Discord server also has a couple custom bots in it. These are:
+
+birb and H0L0's bot [Northstar Servers](https://github.com/hummusbird/northstar-bot), which allows you to run commands like `.status` and see how many people are online on Northstar currently, and things such as `.search region aus`, which would search servers with the region Australia. You can check the repo for more info.
+
+Cyn's bot [Spectre](https://github.com/itscynxx/Spectre) automatically replies to specific error messages, which you can see a list of [here](https://github.com/itscynxx/Spectre/wiki/Replies)
```
Could make subsections for those or a list so that it's formatted in a more readable way \^\^

GeckoEidechse replied:
Done with 54c9ebda40176bfe23836bb45890b881b4f5f61a

GeckoEidechse commented on docs/helping.md:
```diff
@@ -3,3 +3,85 @@
+
+In the Northstar Discord server, a massively used device to help end users is the tickets bot. A lot of the common issues and questions that come up have tags for them on the Discord server. 
+
+Tags are bits of information that you can easily access by using a slash command. We use them mostly for common questions (such as the tag explaining how you can spawn the double barrel, and that you can't spawn it in most servers) and for linking directly to specific pages on the wiki (such as the section about [Intel gen 10+ cpus](https://r2northstar.gitbook.io/r2northstar-wiki/installing-north ...
```
Newline, missing word, grammar
```suggestion
Tags are bits of information that you can easily access by using a slash command.
We use them mostly for common questions (such as the tag explaining how you can spawn the double barrel shotgun and that you can't spawn it in most servers), and for linking directly to specific pages on the wiki (such as the section about [Intel gen 10+ cpus](https://r2northstar.gitbook.io/r2northstar-wiki/installing-northstar/troubleshooting#intel)).
```

GeckoEidechse commented on docs/helping.md:
```diff
@@ -3,3 +3,85 @@
+
+This entire section only applies to users with `ticket-staff`+ roles who can see the ticket channels in the Discord server
+
+The tickets have a fairly straightforward way of operating. When a user clicks on a button to open a ticket they're greeted with the following screen:\
```
newline in source
```suggestion
The tickets have a fairly straightforward way of operating.
When a user clicks on a button to open a ticket they're greeted with the following screen:\
```

GeckoEidechse commented on docs/helping.md:
```diff
@@ -3,3 +3,85 @@
+`reason` is a box to input the reason for a close request.\
+Using "ac" as described later lets other people know how long until the ticket will close ("ac" meaning **A**uto **C**lose). This helps yourself and others when looking at tickets because the tickets bot doesn't show when a ticket is set to auto close.
+
+If the ticket seems resolved/the user has said they have no issues, use `/closerequest` with `close_delay` set to `8` and the `reason` set to "resolved, ac 8hr"
```
I'd say more like 12 hours to give them a bit more time to come back if need be.

```suggestion
If the ticket seems resolved/the user has said they have no issues, use `/closerequest` with `close_delay` set to `12` and the `reason` set to "resolved, ac 12hr"
```

GeckoEidechse commented on docs/helping.md:
```diff
@@ -3,3 +3,85 @@
+
+You can also use `/closerequest` without giving a `close_delay` or a `reason`, however you should always try to give a reason for closing a ticket, and if you don't give a `close_delay` there's a high chance that the ticket doesn't get closed (quite a few end users don't accept or see the close request)
+
+You can view information on the closed tickets, such as open and close date, who opened the ticket, who closed the ticket, and the reason for closing the ticket in the `#transcripts` channel. The bot also provides an online log of the entire ticket connected to the embed when a ticket is closed.
```
newline in source

```suggestion
You can view information on the closed tickets, such as open and close date, who opened the ticket, who closed the ticket, and the reason for closing the ticket in the `#transcripts` channel.
The bot also provides an online log of the entire ticket connected to the embed when a ticket is closed.
```

GeckoEidechse reviewed (approved):
Honestly, I think we should condense the information on the helping page a bit more so that it doesn't feel like a full on novel.

Should be good enough for now however.

GeckoEidechse commented:
CI failed cause Arch wiki is down :trollface:

GeckoEidechse commented on docs/helping.md:
```diff
@@ -3,3 +3,88 @@
+
+### Spectre
+
+A Discord bot created by Cyn with help from multiple people designed to automatically reply to specific errors messages, screenshots of error messages, and some more quality of life features for helpers. You can check the [GitHub repo](https://github.com/itscynxx/Spectre) for more info on Spectre.
```
Newline in source

```suggestion
A Discord bot created by Cyn with help from multiple people designed to automatically reply to specific errors messages, screenshots of error messages, and some more quality of life features for helpers.
You can check the [GitHub repo](https://github.com/itscynxx/Spectre) for more info on Spectre.
```

GeckoEidechse commented on docs/helping.md:
```diff
@@ -3,3 +3,88 @@
+
+### Northstar Servers
+
+A Discord bot created by birb and rewritten by H0L0 which allows you to run commands like `.status` and see how many people are currently online on Northstar, as  well as commands such as `.search region aus` (which would search for Northstar servers with the region set to Australia). You can check the [GitHub repo](https://github.com/hummusbird/northstar-bot) for more info on Northstar Servers.
```
Newline in source

```suggestion
A Discord bot created by birb and rewritten by H0L0 which allows you to run commands like `.status` and see how many people are currently online on Northstar, as  well as commands such as `.search region aus` (which would search for Northstar servers with the region set to Australia).
You can check the [GitHub repo](https://github.com/hummusbird/northstar-bot) for more info on Northstar Servers.
```
